generator.go

```diff
@@ -161,18 +188,24 @@ func (config *FluentConfigGenerator) AddOutput(name string, tag string, options
 
 // WriteFluentdConfig outputs the config in Fluentd syntax
 func (config *FluentConfigGenerator) WriteFluentdConfig(wr io.Writer) error {
-	tmpl, err := template.New("fluent.conf").Parse(fluentDConfigTemplate)
+	tmpl, err := template.New("fluent.conf").Parse(config.fluentdTemplate)
 	if err != nil {
 		return err
 	}
+	if config.Custom != nil {
+		tmpl.Option("missingkey=error")
+	}
 	return tmpl.Execute(wr, config)
 }
 
 // WriteFluentBitConfig outputs the config in Fluent Bit syntax
 func (config *FluentConfigGenerator) WriteFluentBitConfig(wr io.Writer) error {
-	tmpl, err := template.New("fluent-bit.conf").Parse(fluentBitConfigTemplate)
+	tmpl, err := template.New("fluent-bit.conf").Parse(config.fluentBitTemplate)
 	if err != nil {
 		return err
 	}
+	if config.Custom != nil {
+		tmpl.Option("missingkey=error")
```

Comment:
what does this do?

Reply:
https://golang.org/pkg/text/template/#Template.Option

When `tmpl.Execute` is run, it will now throw an error if the user has failed to provide a key to be mapped. In our case here, this is propagated to the return of `WriteFluentdConfig`/`WriteFluentBitConfig`, which the caller can inspect and, optionally, stop their execution there.

It should help people find bugs in their templating.